tools/cp-caps/rcp_caps_test.py

```diff
@@ -0,0 +1,414 @@
+
+            ret = True if dut_stats['sent_packets'] == packets and ref_stats['received_packets'] > threshold else False
+        else:
+            ret = False
+
+        self.__output_results(commands, ret)
+
+    def __test_diag_repeat(self):
+        delay = 10
+        threshold = 80
+        length = 64
+        channel = 20
+        cmd_diag_repeat = f'diag repeat {delay} {length}'
+        cmd_diag_repeat_stop = 'diag repeat stop'
+        commands = [cmd_diag_repeat, 'diag repeat stop', 'diag stats', 'diag stats clear']
+
+        if self.__support_commands(commands):
+            self.__dut.diag_set_channel(channel)
+            self.__ref.diag_set_channel(channel)
+            self.__ref.diag_radio_receive()
+
+            self.__dut.diag_stats_clear()
+            self.__ref.diag_stats_clear()
+
+            self.__dut.diag_repeat(delay, length)
+            self.__dut.wait(1)
+            self.__dut.diag_repeat_stop()
+            dut_stats = self.__dut.diag_get_stats()
+            ref_stats = self.__ref.diag_get_stats()
+
+            ret = True if dut_stats['sent_packets'] > threshold and ref_stats['received_packets'] > threshold else False
+        else:
+            ret = False
+
+        self.__output_format_bool(cmd_diag_repeat, ret)
+        self.__output_format_bool(cmd_diag_repeat_stop, ret)
+
+    def __test_diag_frame(self):
+        packets = 100
+        threshold = 80
+        channel = 20
+        frame = '00010203040506070809'
+        cmd_diag_frame = f'diag frame {frame}'
+        commands = [cmd_diag_frame, f'diag send {packets}', f'diag stats', f'diag stats clear']
+
+        if self.__support_commands(commands):
+            self.__dut.wait(1)
+            self.__dut.diag_set_channel(channel)
+            self.__ref.diag_set_channel(channel)
+            self.__ref.diag_radio_receive()
+
+            self.__dut.diag_stats_clear()
+            self.__ref.diag_stats_clear()
+
+            self.__ref.diag_frame(frame)
+            self.__dut.diag_send(packets, None)
+            self.__dut.wait(1)
+            dut_stats = self.__dut.diag_get_stats()
+            ref_stats = self.__ref.diag_get_stats()
+
+            ret = True if dut_stats['sent_packets'] == packets and ref_stats['received_packets'] > threshold else False
+        else:
+            ret = False
+
+        self.__output_format_bool(cmd_diag_frame, ret)
+
+    def __support_commands(self, commands: List[str]) -> bool:
+        ret = True
+
+        for command in commands:
+            if self.__dut.is_command_supported(command) is False:
+                ret = False
+                break
+
+        return ret
+
+    def __output_results(self, commands: List[str], support: bool):
+        for command in commands:
+            self.__output_format_bool(command, support)
+
+    def __get_dut_diag_power(self) -> int:
+        return int(os.getenv('DUT_DIAG_POWER')) if os.getenv('DUT_DIAG_POWER') else 10
```

Comment:
I think you can simplify the code by specifying the `default` parameter for `os.getenv`.

```
os.getenv('DUT_DIAG_POWER', '0')
```